Code/Framework/AzCore/AzCore/Math/Internal/SimdMathVec1_scalar.inl

```diff
@@ -255,302 +217,258 @@ namespace AZ
         {
             return (arg1 <= arg2) ? (int32_t)0xFFFFFFFF : 0x00000000;
         }
 
-
         AZ_MATH_INLINE bool Vec1::CmpAllEq(Int32ArgType arg1, Int32ArgType arg2)
         {
             return arg1 == arg2;
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::Select(FloatArgType arg1, FloatArgType arg2, FloatArgType mask)
         {
             return (CastToInt(mask) == 0) ? arg2 : arg1;
         }
 
-
         AZ_MATH_INLINE Vec1::Int32Type Vec1::Select(Int32ArgType arg1, Int32ArgType arg2, Int32ArgType mask)
         {
             return (mask == 0) ? arg2 : arg1;
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::Reciprocal(FloatArgType value)
         {
             return 1.0f / value;
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::ReciprocalEstimate(FloatArgType value)
         {
             return Reciprocal(value);
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::Mod(FloatArgType value, FloatArgType divisor)
         {
             return fmodf(value, divisor);
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::Wrap(FloatArgType value, FloatArgType minValue, FloatArgType maxValue)
         {
             return Scalar::Wrap(value, minValue, maxValue);
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::AngleMod(FloatArgType value)
         {
             return Scalar::AngleMod(value);
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::Sqrt(FloatArgType value)
         {
             return sqrtf(value);
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::SqrtEstimate(FloatArgType value)
         {
             return Sqrt(value);
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::SqrtInv(FloatArgType value)
         {
             return 1.0f / sqrtf(value);
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::SqrtInvEstimate(FloatArgType value)
         {
             return SqrtInv(value);
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::Sin(FloatArgType value)
         {
             return sinf(value);
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::Cos(FloatArgType value)
         {
             return cosf(value);
         }
 
-
         AZ_MATH_INLINE void Vec1::SinCos(FloatArgType value, FloatType& sin, FloatType& cos)
         {
             sin = sinf(value);
             cos = cosf(value);
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::Acos(FloatArgType value)
         {
             return acosf(value);
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::Atan(FloatArgType value)
         {
             return atanf(value);
         }
 
-
         AZ_MATH_INLINE Vec1::FloatType Vec1::Atan2(FloatArgType y, FloatArgType x)
         {
             return atan2f(y, x);
         }
 
+        AZ_MATH_INLINE Vec1::FloatType Vec1::ExpEstimate(FloatArgType x)
+        {
+            return exp(x);
+        }
```

Comment:
That's a really good point that I hadn't considered. It absolutely could work on scalar, although it would generate a pretty bad pair of load-hit-stores on pretty much all CPU architectures.

Reply:
Based on the original paper where it isn't implemented as SIMD, it should still be at least 2x faster than calling exp(x)... :)

Reply:
Yes, yes indeed it is 😄 benchmarks added for verification